### DISCUSSION TITLE
save_image with heatmap on layout

### DISCUSSION THREAD

**Discussion by gudeh** [The-OpenROAD-Project/OpenROAD#3001] [Q&A] [answered]
Hi!

I would like to call the TCL command `save_image`  and include the heatmap values on the layout.

I noticed that by manually clicking on the visibility of a heatmap (such as "Placement Density") the image provided by `save_image` will include the heatmap.

My question is how to do this programmatically? I have a TCL script where I save the data with `gui::dump_heatmap` and I would like to also save the image with heat values afterwards. How can I toggle the visibility of heatmaps with my TCL script?

**Comment by maliberty** [comment, marked as answer]
See gui::set_display_controls in https://github.com/The-OpenROAD-Project/OpenROAD/tree/master/src/gui 

Our nightly CI uses https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/scripts/save_images.tcl which is a good reference.

**Reply by gudeh** [reply]
Nice! I noticed the gui::set_display_controls, but couldn't find the name of the control.

This seemed to do the trick: 
`gui::set_display_controls "Heat Maps/Placement Density" visible true`

The link you provided has another example, also.

**Reply by maliberty** [reply]
The names match what you see in the 'Display Control' panel:
![image](https://user-images.githubusercontent.com/761514/224490791-650652e9-c50c-4e6e-9a09-c027cd9d881e.png)
